### PR TITLE
docs(schemas): document the process for adding schemas and follow it for 3.1 series

### PR DESCRIPTION
**Pull request by james-garner-canonical**
#### Description

We want to ensure that `python-libjuju` supports the latest Juju schemas, and that it's clear to `python-libjuju` maintainers and users what versions are supported. This PR adds documentation on how to do this (`docs/CONTRIBUTING.md`), and follows this process for the existing schemas, updating the `3.1` series (`juju/client/SCHEMAS.md` and `juju/client/schemas-juju-*.json`)

#### QA Steps

No changes to generated code are introduced in this PR. `make client` should not produce any changes to the repo state.

Test should all pass.

#### Notes & Discussion

Replaces:
- #1166

Implicitly depends on:
- #1168
Since we add a schema for `3.1.10` here.

### CODE REVIEW

**Review by dimaqq — APPROVED** [review]

**Comment by james-garner-canonical** [comment]
Docs need updating based on changed names/locations in
- https://github.com/juju/python-libjuju/pull/1155

I'll let you decide whether to merge this and then update the docs in #1155, or merge #1155 and then update this before merging, @dimaqq.

**Pull request merged**
